components/eventing-controller/pkg/deployment/publisher_deployment_test.go

```diff
@@ -1,69 +1,242 @@
 package deployment
 
 import (
+	"fmt"
 	"os"
 	"testing"
 
-	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
 	v1 "k8s.io/api/core/v1"
 
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
 )
 
-func Test_GetBEBEnvVars(t *testing.T) {
-	g := NewGomegaWithT(t)
-	envs := map[string]string{
-		// optional
-		"PUBLISHER_REQUESTS_CPU":    "64m",
-		"PUBLISHER_REQUESTS_MEMORY": "128Mi",
-		"PUBLISHER_REQUEST_TIMEOUT": "10s",
+func TestNewNATSDeployment(t *testing.T) {
+	publisherConfig := env.PublisherConfig{
+		RequestsCPU:     "32m",
+		RequestsMemory:  "64Mi",
+		LimitsCPU:       "100m",
+		LimitsMemory:    "128Mi",
+		Image:           "testImage",
+		ImagePullPolicy: "Always",
 	}
-	defer func() {
-		for k := range envs {
-			err := os.Unsetenv(k)
-			g.Expect(err).ShouldNot(HaveOccurred())
-		}
-	}()
-
-	for k, v := range envs {
-		err := os.Setenv(k, v)
-		g.Expect(err).ShouldNot(HaveOccurred())
+	testCases := []struct {
+		name                  string
+		givenBackend          string
+		wantBackendAssertions func(t *testing.T, deployment appsv1.Deployment)
+	}{
+		{
+			name:                  "NATS should be set properly after calling the constructor",
+			givenBackend:          "NATS",
+			wantBackendAssertions: natsBackendAssertions,
+		},
+		{
+			name:                  "BEB should be set properly after calling the constructor",
+			givenBackend:          "BEB",
+			wantBackendAssertions: bebBackendAssertions,
```

Comment:
The test name is `TestNewNATSDeployment` and we are also testing the BEB case in it, so please make it consistent :)